pkg/adaptor/rethinkdb.go

```diff
@@ -58,22 +79,131 @@ func NewRethinkdb(p *pipe.Pipe, path string, extra Config) (StopStartListener, e
 	}
 	r.debug = conf.Debug
 
+	r.client, err = gorethink.Connect(gorethink.ConnectOpts{
+		Address: r.uri.Host,
+		MaxIdle: 10,
+		Timeout: time.Second * 10,
+	})
+	if err != nil {
+		return nil, err
```

Comment:
Go ahead and return `r` instead of `nil` so any logging can happen upstream in the even there are references to the object.